api/internal/handler/route/route.go

```diff
@@ -397,3 +401,61 @@ func Exist(c *gin.Context) (interface{}, error) {
 
 	return nil, nil
 }
+
+type ParamsInput struct {
+	URL          string              `json:"url,omitempty"`
+	BodyParams   map[string]string   `json:"bodyParams,omitempty"`
+	Method       string              `json:"method,omitempty"`
+	HeaderParams map[string][]string `json:"headerParams,omitempty"`
+}
+
+type Result struct {
+	Code    int         `json:"code,omitempty"`
+	Message string      `json:"message,omitempty"`
+	Data    interface{} `json:"data,omitempty"`
+}
+
+func (h *Handler) DebugRequestForwarding(c droplet.Context) (interface{}, error) {
+	paramsInput := c.Input().(*ParamsInput)
+	bodyParams, _ := json.Marshal(paramsInput.BodyParams)
+	client := &http.Client{}
+
+	client.Timeout = 5 * time.Second
+	req, err := http.NewRequest(strings.ToUpper(paramsInput.Method), paramsInput.URL, strings.NewReader(string(bodyParams)))
+	if err != nil {
+		return &data.SpecCodeResponse{StatusCode: http.StatusInternalServerError}, err
+	}
+	for k, v := range paramsInput.HeaderParams {
+		for _, v1 := range v {
+			req.Header.Add(k, v1)
+		}
+	}
+	resp, err := client.Do(req)
+	if err != nil {
+		return &data.SpecCodeResponse{StatusCode: http.StatusInternalServerError}, err
+	}
+	defer func() {
+		if resp != nil {
+			_ = resp.Body.Close()
+		}
+	}()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return &data.SpecCodeResponse{StatusCode: http.StatusInternalServerError}, err
+	}
+	returnData := make(map[string]interface{})
+	result := &Result{}
+	err = json.Unmarshal(body, &returnData)
+	if err != nil {
+		result.Code = resp.StatusCode
+		result.Message = resp.Status
+		result.Data = string(body)
+	} else {
+		result.Code = resp.StatusCode
+		result.Message = resp.Status
+		result.Data = returnData
+
+	}
```

Comment:
We should abstract this so that we could support other protocols such as WebSocket and gRPC in the future.

Reply:
Add a new field `protocol` to distinguish different protocols, and now we only support `http` protocol

Reply:
@liuxiran   
Should define an interface and implement `http`. 
You could refer: 
https://github.com/apache/apisix-dashboard/blob/master/api/internal/core/storage/storage.go
and
https://github.com/apache/apisix-dashboard/blob/master/api/internal/core/storage/etcd.go

Reply:
updated, please recheck when you have time @nic-chen ^_^